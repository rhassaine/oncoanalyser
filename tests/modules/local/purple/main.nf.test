nextflow_process {
    name "Test Process PURPLE"
    script "modules/local/purple/main.nf"
    process "PURPLE"
    config "tests/nextflow.config"

    test("PURPLE") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/sage_vcf/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                output = '/home/rhassaine/container_test_output/subject_a/'
                mode = 'WGS'
            }

            process {
                """
                input[0] = [
                    [id: 'test', tumor_id: 'subject_a.tumor', normal_id: 'subject_a.normal'],
                    file(params.output + '/amber', checkIfExists: true),
                    file(params.output + '/cobalt/', checkIfExists: true),
                    file(params.output + '/esvee/subject_a.tumor.esvee.somatic.vcf.gz', checkIfExists: true),
                    file(params.output + '/esvee/subject_a.tumor.esvee.somatic.vcf.gz.tbi', checkIfExists: true),
                    file(params.output + '/esvee/subject_a.tumor.esvee.germline.vcf.gz', checkIfExists: true),
                    file(params.output + '/esvee/subject_a.tumor.esvee.germline.vcf.gz.tbi', checkIfExists: true),
                    file(params.output + '/pave/subject_a.tumor.sage.somatic.pave.vcf.gz', checkIfExists: true),
                    file(params.output + '/pave/subject_a.tumor.sage.germline.pave.vcf.gz', checkIfExists: true)
                ]
                input[1] = file(params.resource_path + "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna", checkIfExists: true)
                input[2] = 38
                input[3] = file(params.resource_path + "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.fai", checkIfExists: true)
                input[4] = []
                input[5] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/dna/copy_number/GC_profile.1000bp.38.cnp", checkIfExists: true)
                input[6] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/dna/variants/KnownHotspots.somatic.38.vcf.gz", checkIfExists: true)
                input[7] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/dna/variants/KnownHotspots.germline.38.vcf.gz", checkIfExists: true)
                input[8] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/common/DriverGenePanel.38.tsv", checkIfExists: true)
                input[9] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/common/ensembl_data/", checkIfExists: true)
                input[10] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/dna/copy_number/cohort_germline_del_freq.38.csv", checkIfExists: true)
                input[11] = []
                input[12] = []
                input[13] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.purple_dir != null },
                { assert process.out.purple_dir.size() > 0 },
                { 
                    def purpleDir = path(process.out.purple_dir[0][1])
                    assert purpleDir.exists()
                    assert purpleDir.list().size() > 0
                    def somaticTsv = purpleDir.resolve("subject_a.tumor.purple.cnv.somatic.tsv")
                    assert somaticTsv.exists()
                    assert snapshot(somaticTsv.text).match()
                }
            )

            if (process.success) {
                assertAll(
                    { assert path(process.out.purple_dir[0][1]).exists() },
                    { assert path(process.out.purple_dir[0][1]).list().size() > 0 },
                    { assert process.out.versions != null },
                    { assert process.out.versions.size() > 0 },
                    { assert path(process.out.versions[0]).exists() }
                )
            }
        }
    }

    test("PURPLE should fail when input is missing") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/sage_vcf/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = [
                    [id: 'test', tumor_id: 'subject_a.tumor', normal_id: 'subject_a.normal'],
                    [],  // Missing AMBER file
                    [],  // Missing COBALT file
                    [],  // Missing SV tumor VCF
                    [],  // Missing SV tumor TBI
                    [],  // Missing SV normal VCF
                    [],  // Missing SV normal TBI
                    [],  // Missing SMLV tumor VCF
                    []   // Missing SMLV normal VCF
                ]
                """
            }
        }

        then {
            assertAll(
                { assert !process.success },
                { assert process.exitStatus != 0 }
            )
        }
    }
}