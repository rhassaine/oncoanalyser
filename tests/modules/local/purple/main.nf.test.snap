{
    "PURPLE": {
        "content": [
            "chromosome\tstart\tend\tcopyNumber\tbafCount\tobservedBAF\tbaf\tsegmentStartSupport\tsegmentEndSupport\tmethod\tdepthWindowCount\tgcContent\tminStart\tmaxStart\tminorAlleleCopyNumber\tmajorAlleleCopyNumber\nchr1\t1\t123605522\t1.7055\t1\t0.5000\t0.5000\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t1\t0.4770\t1\t1\t0.8527\t0.8527\nchr1\t123605523\t248956422\t1.7698\t2\t0.5147\t0.5565\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.4675\t123605523\t123605523\t0.7849\t0.9849\nchr2\t1\t93139350\t2.5396\t2\t0.5242\t0.5701\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t2\t0.5245\t1\t1\t1.0917\t1.4479\nchr2\t93139351\t242193529\t2.3094\t1\t0.5517\t0.5670\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t1\t0.4840\t93139351\t93139351\t1.0000\t1.3094\nchr3\t1\t92214015\t2.1783\t2\t0.5269\t0.5388\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t2\t0.4130\t1\t1\t1.0047\t1.1736\nchr3\t92214016\t198295559\t2.1003\t2\t0.5265\t0.5823\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.4660\t92214016\t92214016\t0.8773\t1.2230\nchr4\t1\t50726025\t3.1288\t2\t0.5307\t0.6289\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t2\t0.4170\t1\t1\t1.1611\t1.9676\nchr4\t50726026\t190214555\t2.2389\t2\t0.5169\t0.5512\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.2935\t50726026\t50726026\t1.0048\t1.2341\nchr5\t1\t48272853\t1.9382\t2\t0.5156\t0.6004\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t2\t0.3705\t1\t1\t0.7745\t1.1637\nchr5\t48272854\t181538259\t1.5768\t1\t0.5161\t0.6342\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t1\t0.3610\t48272854\t48272854\t0.5768\t1.0000\nchr6\t1\t59191910\t1.5743\t1\t0.5000\t0.5000\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t1\t0.4520\t1\t1\t0.7871\t0.7871\nchr6\t59191911\t170805979\t1.8627\t2\t0.5443\t0.5485\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.4925\t59191911\t59191911\t0.8409\t1.0218\nchr7\t1\t59498943\t1.6003\t2\t0.5167\t0.6281\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t2\t0.3660\t1\t1\t0.5951\t1.0052\nchr7\t59498944\t159345973\t1.6411\t1\t0.5294\t0.6093\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t1\t0.4680\t59498944\t59498944\t0.6411\t1.0000\nchr8\t1\t44955504\t1.8961\t1\t0.5312\t0.5274\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t1\t0.4210\t1\t1\t0.8961\t1.0000\nchr8\t44955505\t145138636\t1.6869\t2\t0.5172\t0.6743\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.3740\t44955505\t44955505\t0.5495\t1.1374\nchr9\t1\t20414426\t1.6151\t2\t0.5490\t0.6198\tTELOMERE\tBND\tBAF_WEIGHTED\t2\t0.3580\t1\t1\t0.6141\t1.0010\nchr9\t20414427\t20415000\t0.0000\t0\t0.0000\t1.0000\tBND\tBND\tSTRUCTURAL_VARIANT\t0\t0.0000\t20414427\t20414427\t0.0000\t0.0000\nchr9\t20415001\t44377362\t1.6000\t0\t0.0000\t0.6162\tBND\tCENTROMERE\tSTRUCTURAL_VARIANT\t0\t0.3716\t20415001\t20415001\t0.6141\t0.9859\nchr9\t44377363\t138394717\t2.4679\t2\t0.5242\t0.5885\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.3515\t44377363\t44377363\t1.0155\t1.4524\nchr10\t1\t40640101\t1.9419\t2\t0.5323\t0.6831\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t2\t0.4645\t1\t1\t0.6154\t1.3265\nchr10\t40640102\t133797422\t2.3317\t2\t0.5252\t0.5704\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.5260\t40640102\t40640102\t1.0016\t1.3301\nchr11\t1\t52751710\t2.0000\t0\t0.0000\t0.5011\tTELOMERE\tCENTROMERE\tUNKNOWN\t0\t0.6143\t1\t1\t0.9979\t1.0021\nchr11\t52751711\t118484974\t2.2958\t2\t0.5252\t0.5635\tCENTROMERE\tBND\tBAF_WEIGHTED\t2\t0.3830\t52751711\t52751711\t1.0021\t1.2937\nchr11\t118484975\t118485000\t0.0000\t0\t0.0000\t1.0000\tBND\tBND\tSTRUCTURAL_VARIANT\t0\t0.0000\t118484975\t118484975\t0.0000\t0.0000\nchr11\t118485001\t135086622\t1.6000\t0\t0.0000\t0.6263\tBND\tTELOMERE\tSTRUCTURAL_VARIANT\t0\t0.4505\t118485001\t118485001\t0.5979\t1.0021\nchr12\t1\t35977329\t1.6436\t1\t0.5000\t0.5000\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t1\t0.4380\t1\t1\t0.8218\t0.8218\nchr12\t35977330\t133275309\t1.7018\t2\t0.5345\t0.6273\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.3770\t35977330\t35977330\t0.6343\t1.0675\nchr13\t1\t17025623\t1.7909\t0\t0.0000\t0.6205\tTELOMERE\tCENTROMERE\tLONG_ARM\t0\t0.0000\t1\t1\t0.6796\t1.1113\nchr13\t17025624\t114364328\t1.7909\t2\t0.5178\t0.6205\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.3710\t17025624\t17025624\t0.6796\t1.1113\nchr14\t1\t17086761\t1.5322\t0\t0.0000\t0.6526\tTELOMERE\tCENTROMERE\tLONG_ARM\t0\t0.0000\t1\t1\t0.5322\t1.0000\nchr14\t17086762\t107043718\t1.5322\t1\t0.5172\t0.6526\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t1\t0.4140\t17086762\t17086762\t0.5322\t1.0000\nchr15\t1\t18362626\t2.1523\t0\t0.0000\t0.5377\tTELOMERE\tCENTROMERE\tLONG_ARM\t0\t0.0000\t1\t1\t0.9949\t1.1574\nchr15\t18362627\t101991189\t2.1523\t2\t0.5086\t0.5377\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.4575\t18362627\t18362627\t0.9949\t1.1574\nchr16\t1\t37295919\t2.1721\t2\t0.5092\t0.5253\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t2\t0.5340\t1\t1\t1.0311\t1.1410\nchr16\t37295920\t90338345\t1.8552\t2\t0.5339\t0.5766\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.4570\t37295920\t37295920\t0.7854\t1.0698\nchr17\t1\t24849829\t1.9456\t2\t0.5592\t0.5140\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t2\t0.5760\t1\t1\t0.9455\t1.0001\nchr17\t24849830\t83257441\t1.9357\t2\t0.5334\t0.5798\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.5330\t24849830\t24849830\t0.8134\t1.1222\nchr18\t1\t18161052\t2.0000\t0\t0.0000\t0.5312\tTELOMERE\tCENTROMERE\tUNKNOWN\t0\t0.0000\t1\t1\t0.9377\t1.0623\nchr18\t18161053\t80373285\t2.1708\t2\t0.5242\t0.5106\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.2945\t18161053\t18161053\t1.0623\t1.1085\nchr19\t1\t25844926\t2.0000\t0\t0.0000\t1.0000\tTELOMERE\tCENTROMERE\tUNKNOWN\t0\t0.0000\t1\t1\t0.0000\t2.0000\nchr19\t25844927\t29815664\t0.0000\t0\t0.0000\t1.0000\tCENTROMERE\tMULTIPLE\tSTRUCTURAL_VARIANT\t0\t0.4378\t25844927\t25844927\t0.0000\t0.0000\nchr19\t29815665\t58617616\t2.5619\t1\t0.5517\t0.6097\tMULTIPLE\tTELOMERE\tBAF_WEIGHTED\t1\t0.4090\t29815665\t29815665\t1.0000\t1.5619\nchr20\t1\t28237289\t1.6547\t2\t0.5333\t0.6201\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t2\t0.4245\t1\t1\t0.6287\t1.0260\nchr20\t28237290\t64444167\t1.6188\t1\t0.5357\t0.6177\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t1\t0.4950\t28237290\t28237290\t0.6188\t1.0000\nchr21\t1\t11890183\t1.9233\t0\t0.0000\t0.5273\tTELOMERE\tCENTROMERE\tLONG_ARM\t0\t0.0000\t1\t1\t0.9092\t1.0141\nchr21\t11890184\t46709983\t1.9233\t2\t0.5403\t0.5273\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.3490\t11890184\t11890184\t0.9092\t1.0141\nchr22\t1\t14004552\t1.3156\t0\t0.0000\t0.7729\tTELOMERE\tCENTROMERE\tLONG_ARM\t0\t0.6845\t1\t1\t0.2987\t1.0169\nchr22\t14004553\t50818468\t1.3156\t2\t0.5339\t0.7729\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.5375\t14004553\t14004553\t0.2987\t1.0169\nchrX\t1\t60509060\t2.6931\t1\t0.5161\t0.6287\tTELOMERE\tCENTROMERE\tBAF_WEIGHTED\t1\t0.5140\t1\t1\t1.0000\t1.6931\nchrX\t60509061\t156040895\t2.3094\t2\t0.5259\t0.5709\tCENTROMERE\tTELOMERE\tBAF_WEIGHTED\t2\t0.3535\t60509061\t60509061\t0.9909\t1.3185\n"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-18T06:53:31.290565768"
    }
}